test/tools/cli/IonProcess.java

```diff
@@ -183,79 +210,411 @@ private static void processEvent(IonWriter ionWriterForOutput,
+                ionReader.next();
+                processToEventFromIonStream(ionWriterForOutput,
+                        ionWriterForErrorReport, ionReader, curTable, currentInfo);
 
                 //write a Container_End event and step out
-                EVENT_INDEX = new Event(EventType.CONTAINER_END, curType, null, null,
+                new Event(EventType.CONTAINER_END, curType, null, null,
                         null, null, null, curDepth)
-                        .writeOutput(ionWriterForOutput, ionWriterForErrorReport, CURRENT_FILE, EVENT_INDEX);
+                        .writeOutput(ionWriterForOutput, ionWriterForErrorReport, currentInfo);
                 ionReader.stepOut();
             } else {
-                EVENT_INDEX = ionStreamToEvent(EventType.SCALAR, ionReader)
-                        .writeOutput(ionWriterForOutput, ionWriterForErrorReport, CURRENT_FILE, EVENT_INDEX);
+                ionStreamToEvent(EventType.SCALAR, ionReader)
+                        .writeOutput(ionWriterForOutput, ionWriterForErrorReport, currentInfo);
+            }
+        } while (ionReader.next() != null);
+    }
+
+    //
+    //  functions for processing from EventStream
+    //
+
+    private static void processFromEventStream(IonWriter ionWriterForOutput,
+                                               IonWriter ionWriterForErrorReport,
+                                               IonReader ionReader,
+                                               CurrentInfo currentInfo,
+                                               ProcessArgs args) throws IOException {
+        if (args.getOutputFormat() == OutputFormat.EVENTS) {
+            processToEventStreamFromEventStream(ionWriterForOutput, ionWriterForErrorReport,
+                    ionReader, currentInfo, args);
+        } else {
+            processToNormalFromEventStream(ionWriterForOutput, ionWriterForErrorReport,
+                    ionReader, currentInfo, args);
+        }
+    }
+
+    private static void processToEventStreamFromEventStream(IonWriter ionWriterForOutput,
+                                                      IonWriter ionWriterForErrorReport,
+                                                      IonReader ionReader,
+                                                      CurrentInfo currentInfo,
+                                                      ProcessArgs args) throws IOException {
+        try {
+            while (ionReader.next() != null) {
+                Event event = ionStreamToEvent(ionReader);
+                event.validate();
+                currentInfo.setLastEventType(event.getEventType());
+
+                event.writeOutput(ionWriterForOutput, ionWriterForErrorReport, currentInfo);
+
+                //update eventIndex
+                currentInfo.setEventIndex(currentInfo.getEventIndex() + 1);
+            }
+        } catch (IonException | NumberFormatException | IllegalStateException e) {
```

Comment:
Sure.
The reason I left them here is because that I tried to separate WRITE type and READ type. Now I add a new variable to ProcessContext which called state,